core/Fetch.cpp

```diff
@@ -48,9 +50,49 @@ namespace olympia
                                                          workload->getValueAsString(),
                                                          skip_nonuser_mode_);
 
+        // Setup BTB
+        btb_ = getContainer()->getChild("btb")->getResourceAs<olympia::BTB>();
+
         fetch_inst_event_->schedule(1);
     }
 
+
+    bool Fetch::predictInstruction_(InstPtr inst)
+    {
+        if (btb_->isHit(inst->getPC()))
+        {
+            btb_hits_++;
```

Comment:
Use pre-increment to avoid the copy (although the compiler will do that for you).